test/OrchardCore.Tests/Apis/Context/BlogContext.cs

```diff
@@ -17,6 +18,20 @@ static BlogContext()
         public override async Task InitializeAsync()
         {
             await base.InitializeAsync();
+            await UsingTenantScopeAsync(async scope =>
+            {
+                var shellFeaturesManager = scope.ServiceProvider.GetRequiredService<IShellFeaturesManager>();
+
+                var features = await shellFeaturesManager.GetAvailableFeaturesAsync();
+
+                // Enable the 'OrchardCore.ContentFields.Indexing.SQL' feature to test out MySQL index lengths.
+                var feature = features.FirstOrDefault(f => f.Id == "OrchardCore.ContentFields.Indexing.SQL");
+
+                if (feature != null)
+                {
+                    await shellFeaturesManager.EnableFeaturesAsync(new[] { feature }, true);
+                }
+            });
```

Comment:
In the `SiteStartup` used for tests, we already call `AddTenantFeatures()` for the same kind of reason, so I also thought about just adding this feature if that works, otherwise try with `AddGlobalFeatures()`. Hmm but yes this would imply that all contexts (not only `BlogContext`) would have this feature enabled, but maybe not a problem.

                .AddTenantFeatures(
                    "OrchardCore.Apis.GraphQL",
                    "OrchardCore.ContentFields.Indexing.SQL"
                )

We could also think about light recipes only used for testings (here only a feature step) that we could update as needed.

Reply:
@jtkech are you suggesting to use SiteStartup or leave it the way I did it?

Reply:
Only a suggestion but finally I think that having a recipe dedicated for tests would be better.

This way we could do a test with all features enabled as suggested by @sebastienros.

Reply:
As we do to run in tests the `$"blog.lucene.query{RecipesConstants.RecipeExtension}"` recipe.

Reply:
Do we have a way define test only recipes in the test project? If we create recipes for testing, I want to make sure they are not discoverable by the web project

Reply:
I will look at it, I'm watching FRANCE - SOUTH AFRICA ;)

Reply:
For tests we already register a custom recipe harvester.

        collection.AddScoped<IRecipeHarvester, TestRecipeHarvester>();

Defined like this


        public Task<IEnumerable<RecipeDescriptor>> HarvestRecipesAsync()
            => HarvestRecipesAsync(new[]
            {
                "Apis/Lucene/Recipes/luceneQueryTest.json"
            });
            ...
            ...

We could define and register a testing recipe including all features, unless those needing external services, and then have a dedicated test that setup a tenant using this recipe.

Reply:
@jtkech i'll try to spend time on this tomorrow is time permits. Meanwhile, if you want to make a change, feel free to modify this PR.

Reply:
No problem, will look at it when I will have time this night.